pkg/controller/infinispan/util/cluster.go

```diff
@@ -153,6 +157,62 @@ func (c Cluster) ExistsCache(cacheName, secretName, podName, namespace, protocol
 	}
 }
 
+// ExistsCacheWithAuth returns true if cacheName cache exists on the podName pod
+func (c Cluster) ExistsCacheWithAuth(user, pass, cacheName, podName, namespace, protocol string) (bool, error) {
+	podIP, err := c.Kubernetes.GetPodIP(podName, namespace)
+	if err != nil {
+		return false, err
+	}
+
+	httpURL := fmt.Sprintf("%s://%s:11222/rest/v2/caches/%s?action=config", protocol, podIP, cacheName)
+	commands := []string{"curl",
+		"--insecure",
+		"-o", "/dev/null", "-w", "%{http_code}", // ignore output and get http status code
+		"-u", fmt.Sprintf("%v:%v", user, pass),
+		"--head",
+		httpURL,
+	}
+
+	execOptions := ExecOptions{Command: commands, PodName: podName, Namespace: namespace}
+	execOut, _, err := c.Kubernetes.ExecWithOptions(execOptions)
+	if err != nil {
+		return false, err
+	}
+
+	httpCode, err := strconv.ParseUint(execOut.String(), 10, 64)
+	if err != nil {
+		return false, err
+	}
+
+	switch httpCode {
+	case 200:
+		return true, nil
+	default:
+		return false, nil
+	}
+}
+
+// CacheNames return the names of the cluster caches available on the pod `podName`
+func (c Cluster) CacheNames(user, pass, podName, namespace, protocol string) ([]string, error) {
+	podIP, err := c.Kubernetes.GetPodIP(podName, namespace)
+	if err != nil {
+		return nil, err
+	}
+	httpURL := fmt.Sprintf("%s://%s:11222/rest/v2/caches", protocol, podIP)
+	commands := []string{"curl", "--insecure", "-u", fmt.Sprintf("operator:%v", pass), httpURL}
```

Comment:
@rigazilla please change with user instead of operator constant

Reply:
@rigazilla sorry for disturb, I don't remember if you decide to update this  in future PR as we have very long discussion

Reply:
@dmvolod yeah I'll clean up all the methods in a next refactoring PR. 
This PR is trying to get out of control :)